core/instance.go

```diff
@@ -504,6 +497,20 @@ func (i *instance) generateTagsList() []*ec2.TagSpecification {
 		},
 	}
 
+	if i.asg.LaunchTemplate != nil {
+		tags.Tags = append(tags.Tags, &ec2.Tag{
+			Key:   aws.String("LaunchTemplateName"),
+			Value: i.asg.LaunchTemplate.LaunchTemplateName,
+		})
+	}
+
+	if i.asg.LaunchConfigurationName != nil {
+		tags.Tags = append(tags.Tags, &ec2.Tag{
+			Key:   aws.String("LaunchConfigurationName"),
+			Value: i.asg.LaunchConfigurationName,
+		})
+	}
```

Comment:
Logic wise wouldn't that be an `if / else if` - rather than 2 if?

Reply:
AWS will make sure an ASG can only have one or the other, so the if/else or if/if will act the same, from what I have seen. That being said, if you prefer if/else I can fix the PR?

Reply:
I mean it's nothing big, but as the PR will be re-updated for adding tests, then I guess it could be updated.

As you said:
> AWS will make sure an ASG can only have one or the other [...]

It's indeed purely from a logical point of view, if it can only be one of them then better add an `if / else if` to make it explicit, rather than 2 if which somehow imply that both could happen.

Reply:
I agree.